ndk/src/media/media_codec.rs

```diff
@@ -400,18 +657,31 @@ impl MediaCodec {
     }
 
     pub fn release_output_buffer(&self, buffer: OutputBuffer<'_>, render: bool) -> Result<()> {
+        debug_assert!(ptr::eq(self, buffer.codec));
+        self.release_output_buffer_by_index(buffer.index, render)
+    }
+
+    pub fn release_output_buffer_by_index(&self, buffer_index: usize, render: bool) -> Result<()> {
         let status =
-            unsafe { ffi::AMediaCodec_releaseOutputBuffer(self.as_ptr(), buffer.index, render) };
+            unsafe { ffi::AMediaCodec_releaseOutputBuffer(self.as_ptr(), buffer_index, render) };
         MediaError::from_status(status)
     }
 
     pub fn release_output_buffer_at_time(
         &self,
         buffer: OutputBuffer<'_>,
         timestamp_ns: i64,
+    ) -> Result<()> {
+        self.release_output_buffer_at_time_by_index(buffer.index, timestamp_ns)
```

Comment:
Really nitty, but let's be consistent.


```suggestion
         debug_assert!(ptr::eq(self, buffer.codec));
         self.release_output_buffer_at_time_by_index(buffer.index, timestamp_ns)
```